docs/source/compression/compression_config_list.rst

```diff
@@ -1,11 +1,190 @@
+Only module types in this list can be selected to be compressed.
+If this key is not set, all module types can be selected.
+
+op_names
+^^^^^^^^
+
+A list of fully-qualified name of modules that will be compressed.
+If the name referenced module is not existed in the model, it will be ignored.
+
+op_names_re
+^^^^^^^^^^^
+
+A list of regular expressions for matching module names.
+The matched modules will be selected to be compressed.
+
+exclude_op_types
+^^^^^^^^^^^^^^^^
+
+A list of type names of classes that inherit from ``torch.nn.Module``.
+Module types in this list can not be selected to be compressed.
+
+exclude_op_names
+^^^^^^^^^^^^^^^^
+
+A list of fully-qualified name of modules that will be excluded.
+
+exclude_op_names_re
+^^^^^^^^^^^^^^^^^^^
+
+A list of regular expressions for matching module names.
+The matched modules will be removed from the modules that need to be compressed.
+
+target_names
+^^^^^^^^^^^^
+
+A list of legal compressed target name, i.e., usually ``_input_``, ``weight``, ``bias``, ``_output_`` are support to be compressed.
+
+Two kinds of target are supported by design, module inputs/outputs(should be a tensor), module parameters/buffers:
+
+- Inputs/Outputs: If the module inputs or outputs is a singal tensor, directly set ``_input_`` for input and ``_output_`` for output.
+  ``_input_{position_index}`` or ``_input_{arg_name}`` can be used to specify the input target,
+  i.e., for a forward function ``def forward(self, x: Tensor, y: Tensor, z: Any): ...``, ``_input_0`` or ``_input_x`` can be used to specify ``x`` to be compressed,
+  note that ``self`` will be ignored when count the position index.
+  Similarly, ``_output_{position_index}`` can be used to specify the output target if the output is a ``list/tuple``,
+  ``_output_{dict_key}`` can be used to specify the output target if the output is a ``dict``.
+- Parameters/Buffers: Directly using the attribute name to specify the target, i.e., ``weight``, ``bias``.
+
+target_settings
+^^^^^^^^^^^^^^^
+
+A ``dict`` of target settings, the format is ``{target_name: setting}``. Target setting usually configure how to compress the target.
+
+All other keys(except these six common keys) in a config will seems as a shortcut of setting key, and will apply to all targets selected in this config.
+
+.. Note:: Each compression target can only be configure once, re-configuration will not take effect.
+
+Pruning Setting Keys
+--------------------
+
+sparse_ratio
+^^^^^^^^^^^^
+
+A float number between 0. ~ 1., the sparse ratio of the pruning target or the overall sparse ratio group of pruning tagrets.
+For example, if the sparse ratio is 0.8, and the pruning target is a Linear module weight, 80% weight value will be masked after pruning.
+
+max_sparse_ratio
+^^^^^^^^^^^^^^^^
+
+A float number between 0. ~ 1., for each pruning target, the sparse ratio after pruning will not be larger than this number,
+that means at most masked ``max_sparse_ratio`` pruning target value.
```

Comment:
max_sparse_ratio is for each single pruning target or for a group of pruning targets?

Reply:
for each single pruning target